tests/library/client-certificates.spec.ts

```diff
@@ -302,6 +303,47 @@ test.describe('browser', () => {
     await page.close();
   });
 
+  test('should not hang on tls errors during TLS 1.2 handshake', async ({ browser, asset, browserName }) => {
+    for (const tlsVersion of ['TLSv1.3', 'TLSv1.2'] as const) {
+      await test.step(`TLS version: ${tlsVersion}`, async () => {
+        const server = tls.createServer({
```

Comment:
I didn't want to pollute our existing server fixture with more specific options which are only relevant for this test, hence creating a pure TLS server here.